RESTserver/src/main/java/com/softserve/if072/restservice/service/AnalyticsService.java

```diff
@@ -65,8 +67,90 @@ public ProductStatistics getProductStatistics(int productId) throws NotEnoughDat
             int purchasingDatesSize = productStatistics.getPurchasingProductDates().length;
             productStatistics.setLastPurchasingDate(productStatistics.getPurchasingProductDates()[purchasingDatesSize - 1]);
         }
+        prepareDataForCharts(productStatistics);
         LOGGER.info(successfullyOperation, productId);
 
         return productStatistics;
     }
+
+    private static void prepareDataForCharts(ProductStatistics productStatistics) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        for (double productUsingSpeed : productStatistics.getProductUsingSpeeds()) {
+            stringBuilder.append(String.format("%.1f ", productUsingSpeed));
+        }
+        String stringData = stringBuilder.toString().trim();
+        stringData = stringData.replace(',', '.');
+        productStatistics.setProductUsingSpeedsForChart(stringData);
+        productStatistics.setProductUsingSpeeds(null);
+
+        convertAmountsAndDatesDataFromArrayToString(productStatistics, Action.USED);
+        convertAmountsAndDatesDataFromArrayToString(productStatistics, Action.PURCHASED);
+    }
+
+    private static void convertAmountsAndDatesDataFromArrayToString(ProductStatistics productStatistics, Action action) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)